commit d25e28d584d20650127cde9b1ea66c41eaf187f2
Author: Nikhil Benesch <nikhil.benesch@gmail.com>
Date:   Mon Apr 3 11:19:29 2017 -0700

    replace sometimes-undefined uint type with unsigned int
    
    Summary:
    `uint` is nonstandard and not a built-in type on all compilers; replace it
    with the always-valid `unsigned int`. I assume this went unnoticed because
    it's inside an `#ifdef ROCKDB_JEMALLOC`.
    Closes https://github.com/facebook/rocksdb/pull/2075
    
    Differential Revision: D4820427
    
    Pulled By: ajkr
    
    fbshipit-source-id: 0876561

diff --git a/db/db_impl.cc b/db/db_impl.cc
index ce38b471..3aaceb95 100644
--- a/db/db_impl.cc
+++ b/db/db_impl.cc
@@ -607,7 +607,7 @@ static void GetJemallocStatus(void* mstat_arg, const char* status) {
 static void DumpMallocStats(std::string* stats) {
 #ifdef ROCKSDB_JEMALLOC
   MallocStatus mstat;
-  const uint kMallocStatusLen = 1000000;
+  const unsigned int kMallocStatusLen = 1000000;
   std::unique_ptr<char[]> buf{new char[kMallocStatusLen + 1]};
   mstat.cur = buf.get();
   mstat.end = buf.get() + kMallocStatusLen;
